docs/database/0005_now_playing_content.sql

```diff
@@ -0,0 +1,72 @@
+CREATE TABLE now_playing_content_base (
+    id SERIAL PRIMARY KEY,
+    account_id INTEGER NOT NULL REFERENCES account(id) ON DELETE CASCADE,
+    playback_position numeric_20_11 NOT NULL
+);
+
+CREATE TABLE now_playing_content_item (
+    item_id INTEGER NOT NULL REFERENCES item(id) ON DELETE CASCADE,
+    UNIQUE (account_id)
+) INHERITS (now_playing_content_base);
+
+CREATE TABLE now_playing_content_item_add_by_rss (
+    item_data jsonb NOT NULL,
+    UNIQUE (account_id)
+) INHERITS (now_playing_content_base);
+
+CREATE TABLE now_playing_content_item_chapter (
+    item_chapter_id INTEGER NOT NULL REFERENCES item_chapter(id) ON DELETE CASCADE,
+    UNIQUE (account_id)
+) INHERITS (now_playing_content_base);
+
+CREATE TABLE now_playing_content_clip (
+    clip_id INTEGER NOT NULL REFERENCES clip(id) ON DELETE CASCADE,
+    UNIQUE (account_id)
+) INHERITS (now_playing_content_base);
+
+CREATE TABLE now_playing_content_item_soundbite (
+    soundbite_id INTEGER NOT NULL REFERENCES item_soundbite(id) ON DELETE CASCADE,
+    UNIQUE (account_id)
+) INHERITS (now_playing_content_base);
+
+/* Ensure that an account can only have one of any type assigned to it at a time */
+
+CREATE OR REPLACE FUNCTION check_account_id_uniqueness()
+RETURNS TRIGGER AS $$
+BEGIN
```

Comment:
I think this is a good use case for a UNION instead of a JOIN here. Nice work.

Reply:
Thank you :)

That said...we are actually discussing possibly merging all 3 of these tables...

history
now_playing
queue

And making them one table, where the position would be negative for history items, 0 for now playing item, and positive for upcoming queue items.

If that's the case, then I think we may not need a union like this.